File: src/main/java/com/uwetrottmann/trakt/v2/services/Genres.java
package com.uwetrottmann.trakt.v2.services;

import com.uwetrottmann.trakt.v2.entities.Genre;
import retrofit.http.GET;

import java.util.List;

public interface Genres {

    /**
     * Get a list of all genres for shows, including names and slugs.
     */
    @GET("/genres/movies")
    List<Genre> movies();

    /**
     * Get a list of all genres for movies, including names and slugs.
     */
    @GET("/genres/shows")
    List<Genre> shows();

}

File: src/main/kotlin/com/redissi/trakt/services/Genres.kt
package com.redissi.trakt.services

import com.redissi.trakt.entities.Genre
import retrofit2.http.GET

interface Genres {
    /**
     * Get a list of all genres for shows, including names and slugs.
     */
    @GET("genres/movies")
    suspend fun movies(): List<Genre>?

    /**
     * Get a list of all genres for movies, including names and slugs.
     */
    @GET("genres/shows")
    suspend fun shows(): List<Genre>?
}